workflow/airflow_libero/operators.py

```diff
@@ -0,0 +1,40 @@
+import json
+from typing import Any, Dict
+
+from airflow.operators.python_operator import PythonOperator
+
+from .event_utils import (
+    get_channel,
+    get_base_message,
+    TASK_EVENT_EXCHANGE,
+)
+
+
+class EventEmittingPythonOperator(PythonOperator):
```

Comment:
this class is an external integration with RabbitMQ and looks identical to the pure Python case

Reply:
It was the cleanest/quickest way for me to get some events out of the tasks without having to mess with its way of executing. It also has a benefit over a simple decorator as you have access to the dag its being run in.

Would also need to do something similar with the base `DAG` class if we wanted dag events as well (started -- failed -- finished).